src/lib/operators/abstract_operator.cpp

```diff
@@ -24,7 +26,8 @@ AbstractOperator::AbstractOperator(const OperatorType type, const std::shared_pt
                                    const std::shared_ptr<const AbstractOperator>& right,
                                    std::unique_ptr<AbstractOperatorPerformanceData> init_performance_data)
     : performance_data(std::move(init_performance_data)), _type(type), _left_input(left), _right_input(right) {
-  // Tell input operators that we want to consume their output
+  // Tell input operators that we want to consume their output. Operators with uncorrelated subqueries have to call
+  // _search_and_register_subqueries for desired expressions to register as a consumer of the subqueries.
```

Comment:
> `_search_and_register_subqueries` sounds like it's for all operators but it's only for ops with uncorrelated subqueries, right?

Any operator could call it in its constructor (bot nothing would happen for most of them). However, we cannot have the call of `_search_and_register_subqueries` in the constrcutor of `AbstractOperator` since we have no expression(s) to search in at this point.

> Who tells something here? I don't get the comment.

``This operator informs all input operators that it wants to consume their output (so it is not deleted until this operator eventually executes). Operators that use expressions that might contain uncorrelated subqueries have to call `_search_and_register_subqueries` to also register as a consumer of these subqueries (see, e.g., the constructors of TableScan or Projection).``

Reply:
How do I read the “thumbs up”?

Reply:
renamed it according to your suggestion